react/containers/ShowAsset/view.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import ErrorPage from 'components/ErrorPage';
+import ShowAssetLite from 'components/ShowAssetLite';
+import ShowAssetDetails from 'components/ShowAssetDetails';
+
+import { ASSET } from 'constants/show_request_types';
+
+function requestIsAnAssetRequest ({ requestType }) {
+  return requestType === ASSET;
+}
+
+class ShowAsset extends React.Component {
+  componentDidMount () {
+    const { asset, previousRequest, requestId, requestName, requestModifier } = this.props;
+    if (!previousRequest) {  // case: the asset request does not exist
+      return this.props.onNewRequest(requestId, requestName, requestModifier);
+    };
+    if (!asset) {  // case: the asset request does not exist
+      const { name, claimId } = previousRequest;
+      return this.props.onShowNewAsset(name, claimId);
+    };
+  }
+  componentWillReceiveProps (nextProps) {
+    if (requestIsAnAssetRequest(nextProps)) {
+      const { asset, previousRequest, requestId, requestName, requestModifier } = nextProps;
+      if (!previousRequest) {
```

Comment:
I think you are still creating more work than necessary with this. In my opinion `previousRequest` shouldn't even exist. In the `mapStateToProps` you should be able to map the `asset` from your state into the component. If `!asset` then make the request. 

I also think `onShowNewAsset` and `onNewRequest` can be combined. More specifically I don't think `onShowNewAsset` is needed. It might just be my lack of understanding with the current data flow, but you shouldn't need to manually say "show this asset". A better approach would be "select the asset with this id".

Reply:
Ok, I think I'm getting closer.  I was able to do away with `onShowNewAsset` and combine the needed logic from its action (retrieving the asset's claim data) into `onNewRequest`.  That allowed me to remove `previousRequest` from the props I am passing to the `<ShowAsset />` component.  However, I am still checking for a `previousRequest` in the mapStateToProps function.  Do you see a way to avoid that step altogether?  The reason for storing and checking the previous requests is to avoid having to retrieve new information for a request that was already made (i.e. to avoid having to request the full `claimId` from the server).  I'm trying to figure out if that can be skipped or consolidated, but I am not sure how.